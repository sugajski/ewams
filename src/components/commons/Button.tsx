import React, {FC} from 'react';
import {TouchableOpacity, View, Text} from 'react-native';

import ButtonStyles from './styles/ButtonStyles';

interface IButtonProps {
  title: string;
  onPress: () => void;
}

const Button: FC<IButtonProps> = ({title, onPress}) => (
  <TouchableOpacity onPress={onPress}>
    <View style={ButtonStyles.container}>
      <Text style={ButtonStyles.title}>{title}</Text>
    </View>
  </TouchableOpacity>
);

export default Button;
