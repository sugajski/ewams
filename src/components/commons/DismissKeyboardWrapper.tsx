import React, {FC} from 'react';
import get from 'lodash/get';
import {TouchableWithoutFeedback, View, Keyboard} from 'react-native';

import {useKeyboardStatus} from '../../lib/hooks';
import DismissKeyboardWrapperStyles from './styles/DismissKeyboardWrapperStyles';

interface IDismissKeyboardWrapperProps {
  children: React.ReactNode;
  childRef: any;
  onPress?: () => void;
}

const DismissKeyboardWrapper: FC<IDismissKeyboardWrapperProps> = ({
  children,
  childRef,
  onPress,
  ...props
}) => {
  const {isKeyboardVisible} = useKeyboardStatus();

  return (
    <TouchableWithoutFeedback
      onPress={() => {
        if (isKeyboardVisible) {
          Keyboard.dismiss();
        }

        const isDisabled = get(childRef, 'current.props.disabled', false);
        if (childRef && childRef.current && !isDisabled) {
          childRef.current.focus();
        }

        if (onPress) {
          onPress();
        }
      }}
      {...props}>
      <View style={DismissKeyboardWrapperStyles.wrapper}>
        <View style={DismissKeyboardWrapperStyles.wrapper} pointerEvents="none">
          {children}
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default DismissKeyboardWrapper;
