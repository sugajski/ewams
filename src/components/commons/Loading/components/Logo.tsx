import React, {FC} from 'react';
import {View, ViewProps} from 'react-native';

import LogoStyles from './styles/LogoStyles';

const Logo: FC<ViewProps> = () => (
  <View style={LogoStyles.container}>
    <View style={[LogoStyles.square, LogoStyles.a]} />
    <View style={[LogoStyles.square, LogoStyles.b]} />
    <View style={[LogoStyles.square, LogoStyles.c]} />
    <View style={[LogoStyles.square, LogoStyles.d]} />
  </View>
);

export default Logo;
