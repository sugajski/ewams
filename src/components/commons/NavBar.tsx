import React, {useEffect, FC} from 'react';
import {View, Image, TouchableOpacity, BackHandler} from 'react-native';
import {useHistory} from 'react-router-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';

import {Images} from '../../themes';
import NavBarStyles from './styles/NavBarStyles';

interface INavBarProps {
  additionalBackAction: () => void;
}

const NavBar: FC<INavBarProps> = ({additionalBackAction = null}) => {
  const router = useHistory();
  const safeAreaInsets = useSafeAreaInsets();

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', handleBackPressed);
    return () => {
      BackHandler.removeEventListener('hardwareBackPress', handleBackPressed);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleBackPressed = () => {
    if (additionalBackAction) {
      additionalBackAction();
    }
    router.goBack();
    return true;
  };

  return (
    <View style={[NavBarStyles.container, {height: 56 + safeAreaInsets.top}]}>
      <TouchableOpacity onPress={handleBackPressed}>
        <View style={NavBarStyles.backButtonContainer}>
          <Image
            source={Images.back_ic}
            style={NavBarStyles.backButtonIcon}
            resizeMode="stretch"
          />
        </View>
      </TouchableOpacity>
    </View>
  );
};

export default NavBar;
