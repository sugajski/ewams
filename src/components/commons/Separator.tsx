import React, {FC} from 'react';
import {View} from 'react-native';

import SeparatorStyles from './styles/SeparatorStyles';
import {Colors} from '../../themes';

interface ISeparatorProps {
  color?: string;
}

const Separator: FC<ISeparatorProps> = ({color = Colors.white}) => (
  <View style={[SeparatorStyles.separator, {backgroundColor: color}]} />
);

export default Separator;
