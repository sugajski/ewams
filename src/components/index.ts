import OutlinedTextInput from './OutlinedTextInput';
import Record from './Record';
import SendButton from './SendButton';
import SearchHistory from './SearchHistory';

export {OutlinedTextInput, Record, SendButton, SearchHistory};
