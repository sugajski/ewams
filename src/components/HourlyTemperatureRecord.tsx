import React, {FC} from 'react';
import {View, Text} from 'react-native';

import {Separator} from './commons';
import {Localizable} from '../lib';
import HourlyTemperatureRecordStyles from './styles/HourlyTemperatureRecordStyles';

interface IRecord {
  hour: string;
  temperature: string;
}

interface IHourlyTemperatureRecordProps {
  record: IRecord;
  index: number;
}

const HourlyTemperatureRecord: FC<IHourlyTemperatureRecordProps> = ({
  record,
  index,
}) => (
  <View style={HourlyTemperatureRecordStyles.container}>
    <Text style={HourlyTemperatureRecordStyles.text}>
      {index === 0 ? Localizable.t('weatherDetails.now') : record.hour}
    </Text>
    <Separator />
    <Text style={HourlyTemperatureRecordStyles.text}>{record.temperature}</Text>
  </View>
);

export default HourlyTemperatureRecord;
