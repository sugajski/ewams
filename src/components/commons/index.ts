import DismissKeyboardWrapper from './DismissKeyboardWrapper';
import Loading from './Loading/Loading';
import Separator from './Separator';

export {DismissKeyboardWrapper, Loading, Separator};
