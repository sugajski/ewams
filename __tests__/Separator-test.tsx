import React from 'react';
import renderer from 'react-test-renderer';

import {Separator} from '../src/components/commons';

test('Separator renders correctly', () => {
  const tree = renderer.create(<Separator />).toJSON();
  expect(tree).toMatchSnapshot();
});
