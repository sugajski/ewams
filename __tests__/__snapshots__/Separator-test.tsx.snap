// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 1,
        "textShadowColor": "#00000066",
        "textShadowOffset": Object {
          "height": 1,
          "width": -1,
        },
        "textShadowRadius": 6,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
/>
`;
